engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingScheduleServiceImpl.java

```diff
@@ -68,18 +76,52 @@ public void runAtFixedRate(final Duration delay, final Task task) {
             }));
   }
 
-  private void scheduleOnActor(final Runnable task) {
+  private void useActorControl(final Runnable task) {
+    if (actorControl == null) {
+      LOG.debug("ProcessingScheduleService hasn't been opened yet, ignore scheduled task.");
```

Comment:
:question: Is this an expected case, or a failsafe, i.e. should never happen but if it did no biggie? If expect it to never happen but it's probably fine if it does, we could add an `assert` so it breaks when it happens in tests, but ignores in production.

Reply:
With the current state of the production code it doesn't happen, but it could if someone would use the schedule during replay (which is not allowed).

Reply:
assert doesn't work, since our current tests fail then (which check that we ignore that case :sweat_smile:) not sure how to behave here then

Reply:
Ah, I thought it was specifically a case of "this should never ever happen", but if that's not the case then adding an assert is not the right way. And if we're testing that it should never happen, then that's even better than an assert :smile:

Reply:
Yeah I somehow forgot that I actually wrote several tests regarding these cases to prevent it :sweat_smile: